ai_diffusion/model.py

```diff
@@ -245,18 +246,30 @@ async def enqueue_jobs(
         sampling = ensure(input.sampling)
         params.has_mask = input.images is not None and input.images.hires_mask is not None
 
+        queue_mode = self.queue_mode
         for i in range(count):
             next_seed = sampling.seed + i * settings.batch_size
             input = replace(input, sampling=replace(sampling, seed=next_seed))
             params.seed = next_seed
-            job = self.jobs.add(kind, copy(params))
-            await self._enqueue_job(job, input)
+            new_job = self.jobs.add(kind, copy(params))
+            await self._enqueue_job(new_job, input, queue_mode)
+            if queue_mode is QueueMode.replace:
```

Comment:
Managing the jobs here shouldn't be needed, the clients cancel the jobs and the job queue will react to reflect that.

Reply:
oh yeah huh, why did I not do that

Reply:
done